src/Arcus.WebApi.Jobs/AzureServiceBusCloudEventSubscriptionMessagePump.cs

```diff
@@ -0,0 +1,131 @@
+    /// </summary>
+    public abstract class AzureServiceBusCloudEventSubscriptionMessagePump : AzureServiceBusMessagePump<CloudEvent>
+    {
+        private readonly string _subscriptionName;
+        private readonly AzureServiceBusMessagePumpSettings _messagePumpSettings;
+
+        private static readonly JsonEventFormatter JsonEventFormatter = new JsonEventFormatter();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureServiceBusCloudEventSubscriptionMessagePump"/> class.
+        /// </summary>
+        /// <param name="configuration">Configuration of the application</param>
+        /// <param name="serviceProvider">Collection of services that are configured</param>
+        /// <param name="logger">Logger to write telemetry to</param>
+        /// <exception cref="ArgumentNullException">When the <paramref name="serviceProvider"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">When the <paramref name="serviceProvider"/> doesn't have a registered <see cref="AzureServiceBusMessagePumpSettings"/> instance.</exception>
+        protected AzureServiceBusCloudEventSubscriptionMessagePump(
+            IConfiguration configuration,
+            IServiceProvider serviceProvider,
+            ILogger logger) : base(configuration, serviceProvider, logger)
+        {
+            Guard.NotNull(
+                serviceProvider,
+                nameof(serviceProvider),
+                $"Requires a '{nameof(IServiceProvider)}' implementation to retrieve the '{nameof(AzureServiceBusMessagePumpSettings)}'");
+
+            var messagePumpSettings = serviceProvider.GetRequiredService<AzureServiceBusMessagePumpSettings>();
+            Guard.NotNull<AzureServiceBusMessagePumpSettings, ArgumentException>(
+                messagePumpSettings, 
+                $"The '{nameof(serviceProvider)}:{serviceProvider.GetType().Name}' requires to have a non-null '{nameof(AzureServiceBusMessagePumpSettings)}' instance registered");
+            
+            _messagePumpSettings = messagePumpSettings;
+            _subscriptionName = _messagePumpSettings.SubscriptionName;
+        }
+
+        /// <summary>
+        /// Deserializes a raw JSON message body.
+        /// </summary>
+        /// <param name="rawMessageBody">Raw message body to deserialize</param>
+        /// <param name="messageContext">Context concerning the message</param>
+        /// <returns>Deserialized message</returns>
+        protected override CloudEvent DeserializeJsonMessageBody(byte[] rawMessageBody, MessageContext messageContext)
+        {
+            Guard.NotNull(rawMessageBody, nameof(rawMessageBody), "Cannot deserialize raw JSON body from 'null' input");
+            Guard.NotAny(rawMessageBody, nameof(rawMessageBody), "Cannot deserialize raw JSON body from empty input");
+
+            CloudEvent cloudEvent = JsonEventFormatter.DecodeStructuredEvent(rawMessageBody);
```

Comment:
Yeah, because that problem. And also because it's a single event. Otherwise, I have to check the batch of events, and also, we haven't yet a signature to parse a series of bytes, which the official CloudEvents SDK does.
Perf-wise, maybe some conversion saved.

Reply:
Do keep in mind that the topic will just contain what Event Grid is publishing. This means that it can be a list or multiple objects.

In terms of the bytes, might be something we have to add in Event Grid then, no?

Reply:
Yeah, I think both things (support single and multiple) and bytes are something that we may want to add to that repo. But we should (like you said) probably come up with a better approach to structure our parsing logic before introducing this.

Reply:
Let's track an issue in this repo to move to the new approach (and point to event grid issue) to make sure we swap this out when we can then.

Reply:
#104